z3/src/z3_term.cpp

```diff
@@ -63,87 +63,112 @@ bool Z3TermIter::equal(const TermIterBase & other) const
 
 // Z3Term implementation
 
-size_t Z3Term::hash() const
-{
-throw NotImplementedException(
-      "Term iteration not implemented for Z3 backend.");
+size_t Z3Term::hash() const {
+	if (is_function) {
+		return term.hash();
+	} else {
+		return z_func.hash();
+	}
+	return term.hash();
 }
 
-bool Z3Term::compare(const Term & absterm) const
+std::size_t Z3Term::get_id() const
 {
-throw NotImplementedException(
+  throw NotImplementedException(
```

Comment:
Message wasn't changed from copy-paste. Also, if you feel like it, I think Z3 does have a `get_id` method. But it's not that vital.